### PR TITLE
fix: check for nil source repos

### PR DESCRIPTION
**Pull request by plyr4**
closes https://github.com/go-vela/community/issues/612

### CODE REVIEW

File: scm/github/repo.go [hunk 1]
```diff
@@ -346,6 +346,12 @@ func (c *client) ListUserRepos(u *library.User) ([]*library.Repo, error) {
 
 	// iterate through each repo for the user
 	for _, repo := range r {
+		// skip if the repo is void
+		//   fixes an issue with GitHub replication being out of sync
```
Comment (JordanSussman) [inline scm/github/repo.go:350]: Remove extra whitespace
```suggestion
		// fixes an issue with GitHub replication being out of sync
```

**Review by jbrockopp — DISMISSED** [review]
LGTM

File: scm/github/repo.go (hunk 1 above)
Comment (plyr4) [reply scm/github/repo.go]: resolved, thank you

**Review by jbrockopp — APPROVED** [review]
LGTM

**Pull request merged**